x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/index/engine/FollowingEngine.java

```diff
@@ -131,11 +131,6 @@ protected long generateSeqNoForOperationOnPrimary(final Operation operation) {
         return operation.seqNo();
     }
 
-    @Override
```

Comment:
@henningandersen Two reasons that I removed it:

- This should be a noop for the following engine since MSU should be at least the seqno of a delete.
- We should delegate to advanceMaxSeqNoOfUpdatesOrDeletesOnPrimary not only deletes but also updates.

The downside is that we might catch bugs where MSU from the leader is not passed correctly with advanceMaxSeqNoOfUpdatesOrDeletesOnPrimary.

I am good with either keeping or removing it. What do you think?

Reply:
I opt to remove it. One option to mitigate the downside is to override `delete()` and assert that `seqNo <= MSU` in `FollowingEngine`?

Reply:
good plan. We talked about this, Nhat. After reflection, let's do it here in this PR.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/41161/commits/0f4492bc3c9f807559c421e715ad68e852868bb2 to restore this method and add an assertion in the following engine.